### PR TITLE
(#1885143) bind_remount: avoid calling statvfs

### PR DESCRIPTION
**Pull request by dtardon**

### CODE REVIEW

File: src/basic/mount-util.c [hunk 1]
```diff
@@ -313,55 +315,44 @@ int umount_recursive(const char *prefix, int flags) {
          * unmounting them until they are gone. */
 
         do {
-                _cleanup_fclose_ FILE *proc_self_mountinfo = NULL;
+                _cleanup_(mnt_free_tablep) struct libmnt_table *table = NULL;
+                _cleanup_(mnt_free_iterp) struct libmnt_iter *iter = NULL;
 
                 again = false;
                 r = 0;
```
Comment (msekletar) [inline src/basic/mount-util.c]: Nit, assigning to `r` seems redundant.

File: src/basic/mount-util.c [hunk 2]
```diff
@@ -313,55 +315,44 @@ int umount_recursive(const char *prefix, int flags) {
...
+                        path = mnt_fs_get_target(fs);
+                        if (!path)
                                 continue;
 
-                        if (umount2(p, flags) < 0) {
-                                r = log_debug_errno(errno, "Failed to umount %s: %m", p);
+                        if (!path_startswith(path, prefix))
+                                continue;
+
+                        if (umount2(path, flags) < 0) {
```
Comment (msekletar) [inline src/basic/mount-util.c]: In upstream there was a follow-up (827ea5). It figure we should backport it too.

**Review by msekletar — CHANGES_REQUESTED** [review]

File: src/basic/mount-util.c (hunk 1 above)
Comment (dtardon) [reply src/basic/mount-util.c]: Indeed.

File: src/basic/mount-util.c (hunk 2 above)
Comment (dtardon) [reply src/basic/mount-util.c]: Done.

**Review by msekletar — APPROVED** [review]
LGTM

**Pull request merged**